docs/libraries/workflow/GLOSSARY.md

```diff
@@ -1,44 +1,46 @@
-TODO
-
 # Glossary
 
 ## Worker
 
-A process that's running workflows.
+A process that queries for pending workflows with a specific filter. Filter is based on which workflows are registered in the given worker's registry.
+
+## Registry
 
-There are usually multiple workers running at the same time.
+A collection of registered workflows.
 
 ## Workflow
 
-A series of activies to be ran together.
+A series of fallible executions of code (also known as activities), signal listeners, signal transmitters, or sub workflow triggers.
 
-The code defining a workflow only specifies what activites to be ran. There is no complex logic (e.g. database queries) running within workflows.
+Workflows can be though of as a list of tasks. The code defining a workflow only specifies what items should be ran; There is no complex logic (e.g. database queries) running within the top level of the workflow.
 
-Workflow code can be reran multiple times to replay a workflow.
+Upon an activity failure, workflow code can be reran without duplicate side effects because activities are cached and re-read after they succeed.
 
-## Workflow State
+## Activity
 
-Persistated data about a workflow.
+A block of code that can fail. This cannot trigger other workflows or activities, but it can call operations.
 
-## Workflow Run
+## Operation
 
-An instance of a node running a workflow. If re-running a workflow, it will be replaying events.
+A block of code. Can fail or not fail, used simply for tidiness.
```

Comment:
I'm not sure I understand what "tidiness" means in this context. I'm also
curious about the Activity vs Operation distinction. What can an Activity do
that an operation can't?

Does this basically mean that it's more like separating functions? So an example
of an Activity might be "create a new user", and this uses the operations "find
existing user" and "create new user" and "send user email"?